### PR TITLE
First Pass at 4.0.10 -> 5.0.0 upgrade guide

### PR DESCRIPTION
**Pull request by jgrund**
Signed-off-by: Joe Grund <jgrund@whamcloud.io>

### CODE REVIEW

File: docs/Upgrade_Guide/upgrade_iml-4.0-el7_to_iml-5.0-el7.md [hunk 1]
```diff
@@ -7,7 +7,7 @@ The first component in the environment to upgrade is the Integrated Manager for
 1. Download the latest Integrated Manager for Lustre release repo:
 
    ```sh
-      yum-config-manager --add-repo=yum-config-manager --add-repo=https://raw.githubusercontent.com/whamcloud/integrated-manager-for-lustre/v5.0.0/chroma_support.repo
+      yum-config-manager --add-repo=https://raw.githubusercontent.com/whamcloud/integrated-manager-for-lustre/v5.0.0P0/chroma_support.repo
```
Comment (jgrund) [inline docs/Upgrade_Guide/upgrade_iml-4.0-el7_to_iml-5.0-el7.md]: Keep this as it was before, the P0 is temporary.

File: docs/Upgrade_Guide/upgrade_iml-4.0-el7_to_iml-5.0-el7.md [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+
+1. Run `chroma-config setup` to complete the installation.
+
+1. Perform a hard refresh on the browser and verify that IML loads correctly
+
+## Upgrade the Lustre Servers
+
+Lustre server upgrades can be coordinated as either an online roll-out, leveraging the failover HA mechanism to migrate services between nodes and minimize disruption, or as an offline service outage, which has the advantage of usually being faster to deploy overall, with generally lower risk.
+
+The upgrade procedure documented here describes the faster and more reliable approach, which requires that the filesystem be stopped. It assumes that the Lustre servers have been installed in pairs, where each server pair forms an independent high-availability cluster built on Pacemaker and Corosync. Integrated Manager for Lustre deploys these configurations and includes its own resource agent for managing Lustre assets, called `ocf:chroma:Target`. Integrated Manager for Lustre can also configure STONITH agents to provide node fencing in the event of a cluster partition or loss of quorum.
```
Comment (jgrund) [inline docs/Upgrade_Guide/upgrade_iml-4.0-el7_to_iml-5.0-el7.md]: `ocf:chroma:Target` is no longer used in IML 5. We now use stock Lustre RA and clusterlabs ZFS RA

File: docs/Upgrade_Guide/upgrade_iml-4.0-el7_to_iml-5.0-el7.md [hunk 3]
```diff
@@ -0,0 +1,70 @@
+# Upgrading Integrated Manager for Lustre 4.0.10.0 to Lustre {{site.lustre_version}} and Integrated Manager for Lustre {{site.version}}
```
Comment (jgrund) [inline docs/Upgrade_Guide/upgrade_iml-4.0-el7_to_iml-5.0-el7.md]: Should this be 4.0.x?

File: docs/Upgrade_Guide/upgrade_iml-4.0-el7_to_iml-5.0-el7.md [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+
+   ```sh
+   chroma-agent convert_targets
+   ```
+
+## Summary
+
+The filesystem(s) should now be started. Connect a client and verify that it is able to access files on the filesystem.
+
+[top](#upgrading-integrated-manager-for-lustre-40100-to-lustre-2121-and-integrated-manager-for-lustre-5000)
```
Comment (jgrund) [inline docs/Upgrade_Guide/upgrade_iml-4.0-el7_to_iml-5.0-el7.md]: How does this work when the title contains variables?

File: docs/Upgrade_Guide/upgrade_iml-4.0-el7_to_iml-5.0-el7.md [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+
+The first component in the environment to upgrade is the Integrated Manager for Lustre server and software. The manager server upgrade can be conducted without any impact to the Lustre file system services.
+
+1. Download the latest Integrated Manager for Lustre release repo:
+
+   ```sh
+   yum-config-manager --add-repo=https://raw.githubusercontent.com/whamcloud/integrated-manager-for-lustre/v5.0.0/chroma_support.repo
+   ```
+
+1. Install the updated manager via `yum`:
```
Comment (jgrund) [inline docs/Upgrade_Guide/upgrade_iml-4.0-el7_to_iml-5.0-el7.md]: Shouldn't we also update the manager OS?

File: docs/Upgrade_Guide/upgrade_iml-4.0-el7_to_iml-5.0-el7.md (hunk 3 above)
Comment (johnsonw) [reply docs/Upgrade_Guide/upgrade_iml-4.0-el7_to_iml-5.0-el7.md]: That makes more sense. I'll change.

File: docs/Upgrade_Guide/upgrade_iml-4.0-el7_to_iml-5.0-el7.md (hunk 4 above)
Comment (johnsonw) [reply docs/Upgrade_Guide/upgrade_iml-4.0-el7_to_iml-5.0-el7.md]: The variables will not work unfortunately as markdown won't generate links using`.` I have a task in the npm file that checks for broken links and will alert us if it breaks. Though, we should probably add that as a test check. I'm happy to add that if you want.

File: docs/Upgrade_Guide/upgrade_iml-4.0-el7_to_iml-5.0-el7.md (hunk 5 above)
Comment (johnsonw) [reply docs/Upgrade_Guide/upgrade_iml-4.0-el7_to_iml-5.0-el7.md]: We probably should. I'll add a step to do this.

File: docs/Upgrade_Guide/upgrade_iml-4.0-el7_to_iml-5.0-el7.md (hunk 4 above)
Comment (jgrund) [reply docs/Upgrade_Guide/upgrade_iml-4.0-el7_to_iml-5.0-el7.md]: Hrrm, I thought I saw this being done somewhere in the docs.

Comment (jgrund) [reply docs/Upgrade_Guide/upgrade_iml-4.0-el7_to_iml-5.0-el7.md]: Here it is:

https://github.com/whamcloud/Online-Help/blob/d13f4598a657f21924ed26ef0513954f92a66906/docs/Install_Guide/Create_Offline_Repos_10_0.md#L95

File: docs/Upgrade_Guide/upgrade_iml-4.0-el7_to_iml-5.0-el7.md [hunk 6]
```diff
@@ -0,0 +1,129 @@
...
+
+### Backup the Existing configuration.
+
+Prior to commencing the upgrade, it is essential that a backup of the existing configuration is completed. This will enable recovery of the original configuration in the event of a problem occurring during execution of the upgrade.
+
+The following shell script can be used to capture the essential configuration information that is relevant to the Integrated Manager for Lustre software itself:
+
+```bash
+#!/bin/sh
+# EE Integrated Manager for Lustre (IML) server backup script
```
Comment (jgrund) [inline docs/Upgrade_Guide/upgrade_iml-4.0-el7_to_iml-5.0-el7.md]:
```suggestion
# Integrated Manager for Lustre (IML) server backup script
```

**Pull request merged**